src/modules/editTask.js

```diff
@@ -0,0 +1,104 @@
+const todoList = document.querySelector('.todo-list');
+
+const runComplete = (inputEl) => {
+  const localData = JSON.parse(localStorage.getItem('tasks'));
+  localData.forEach((task) => {
+    if (inputEl.checked !== false) {
+      if (Number(inputEl.id) === Number(task.index)) {
+        task.completed = true;
+      }
+    } else {
+      task.completed = false;
+    }
+    localStorage.setItem('tasks', JSON.stringify(localData));
+  });
+};
+
+const toggleComplete = (inputElement) => {
+  if (inputElement.checked === false) {
+    inputElement.parentElement.parentElement.classList.remove('strike-thro');
+    inputElement.parentElement.parentElement.parentElement.classList.remove('active');
+  } else {
+    inputElement.parentElement.parentElement.classList.add('strike-thro');
+    inputElement.parentElement.parentElement.parentElement.classList.add('active');
+  }
+};
+
+const showEditInputs = (liElement) => {
+  const editInput = document.querySelector('.edit-input');
+  if (editInput) {
+    editInput.remove();
+  }
+  const input = document.createElement('input');
+  input.type = 'text';
+  input.name = 'editInput';
+  input.value = liElement.textContent;
+  input.classList.add('edit-input');
+  liElement.appendChild(input);
+  input.focus();
+};
+
+function editTask() {
+  todoList.addEventListener('click', () => {
+    const editEl = document.querySelectorAll('.edit-task');
+    editEl.forEach((edit) => {
+      edit.addEventListener('change', (e) => {
+        e.stopPropagation();
+        if (e.target.tagName === 'INPUT' && e.target.type === 'checkbox') {
+          toggleComplete(e.target);
+          runComplete(e.target);
+        }
+      });
+    });
+  });
+}
+
+function editText() {
+  todoList.addEventListener('click', () => {
+    const editEl = document.querySelectorAll('.edit-task');
+    editEl.forEach((edit) => {
+      edit.addEventListener('click', (e) => {
+        e.stopPropagation();
+        if (e.target.tagName === 'LI') {
+          showEditInputs(e.target);
+        }
+      });
+    });
+  });
+}
+
+function updateText() {
+  const editInput = document.querySelector('.edit-input');
+  const localData = JSON.parse(localStorage.getItem('tasks'));
+
+  if (!editInput) {
+    return;
+  }
+  const newText = editInput.value;
+  if (newText !== '') {
+    const liInput = editInput.parentElement;
+    liInput.textContent = newText;
+    localData.forEach((data) => {
+      data.text = newText;
+      localStorage.setItem('tasks', JSON.stringify(localData));
+    });
+  }
+  editInput.remove();
+}
```

Comment:
- [ ] <b>REQUIRED:</b> As the previous reviewer mentioned, some functionality are not working as expected.
       -  When I edit one item, sometimes I find that everything is edited, and sometimes I find that it edited other items instead.
       - After adding new item, first, it is displaying duplication of the current data and then reloads the page(This is not the best way to do this).
       - As the previous reviewer mentioned, when I mark a task as completed, it must remain marked completed even after I refresh the page. That is why you are using local storage.